Modules/@babylonjs/react-native/shared/BabylonNative.h

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <jsi/jsi.h>
+#include <ReactCommon/CallInvoker.h>
+
+namespace Babylon
+{
+    void Initialize(facebook::jsi::Runtime& jsiRuntime, std::shared_ptr<facebook::react::CallInvoker> jsCallInvoker);
+    void Deinitialize();
```

Comment:
how important do we feel it is for all platforms to use these function definitions? Right now UWP does not have a call invoker. It also needs to specify window types during update view. Could these functions be templated/generics?

Reply:
do we specifically need to hand around this call invoker or could we just hand around the dispatch function? Handing around the dispatch function and allowing the window type to be set when updating the view seems like it would unblock UWP scenarios

---
In reply to: [546838824](https://github.com/BabylonJS/BabylonReactNative/pull/135#discussion_r546838824) [](ancestors = 546838824)

Reply:
We could probably pass an `std::function` instead of a `CallInvoker`. I set it up to pass a `CallInvoker` because:
1. `CallInvoker` does extra work to get JS promise continuations to run. Without it, promise continuations can "hang" indefinitely. We saw this behavior on both iOS and Android, and I expect it is the same on UWP.
2. `CallInvoker` exposes functions to execute work on the JS thread both synchronously and asynchronously. That said, I discovered that the `CallInvoker::invokeSync` function is not actually implemented (it just throws an unsupported exception).

So again, we should be able to switch it to `std::function`, but I suspect if we don't use `CallInvoker` on UWP, we are going to have problems with JS promise continuations.

Reply:
chatted offline, sounds like we are using CallInvoker on UWP, but we don't have access to it. Because we are using CallInvoker, we felt confident that handing around a std::function using the CallInvoker would still work well here

Reply:
@ryantrem feel free to resolve this convesation, i'll keep it open for now while we wait on changes to pass around a std function